src/app/websites/website-services/afterdark/afterdark.service.ts

```diff
@@ -0,0 +1,201 @@
+      throw new Error(`Bad response getting csrf token ${response.statusCode}`);
+    }
+    const match = response.body.match(new RegExp(`name="csrfmiddlewaretoken" value="([^"]+)"`));
+    if (!match) {
+      throw new Error('Could not find csrf token in body');
+    }
+    const csrfToken = match[1];
+    return csrfToken;
+  }
+
+  private async getUserData(profileId: string, csrfToken: string): Promise<any> {
+    const body = JSON.stringify({
+      "operationName": "UserQuery",
+      "variables": {},
+      "query": `query UserQuery {
+        userViewer {
+          user {
+            id
+            username
+            isUserAdmin
+            isProUser
+            isEmailVerified
+            collectionSet {
+              edges {
+                node {
+                  id
+                  title
+                }
+              }
+            }
+          }
+        }
+      }`
+    });
+    var cookies = await getCookies(profileId, this.BASE_URL);
+    const response = await got.gotPostJSON(`${this.BASE_URL}/graphql/`, body, this.BASE_URL, cookies, profileId, {
+      headers: {
+        'content-type': 'application/json',
+        'X-CSRFToken': csrfToken,
+      }
+    });
+    if (response.statusCode !== 200) {
+      throw new Error(`Bad response getting user data ${response.statusCode}`);
+    }
+    const responseBody = JSON.parse(response.body);
+    if (!(responseBody.data || {}).userViewer) {
+      return null;
+    }
+    const collections = ((((responseBody.data || {}).userViewer || {}).user || {}).collectionSet || {}).edges || [];
+    return {
+      'userId': (((responseBody.data || {}).userViewer || {}).user || {}).id || '',
+      'username': (((responseBody.data || {}).userViewer || {}).user || {}).username || '',
+      'collections': collections.map(c => c.node),
+    };
+  }
+
+  public getFolders(profileId: string): Folder[] {
+    // conveniently, collections have the same structure as folders
+    return this._profileManager.getData(profileId, AfterDark.name).collections;
+  }
+
+  public async checkStatus(profileId: string, data?: any): Promise<WebsiteStatus> {
+    const returnValue: WebsiteStatus = {
+      username: null,
+      status: LoginStatus.LOGGED_OUT,
+    };
+    const csrfToken = await this.getCsrfToken(profileId);
+    const userData = await this.getUserData(profileId, csrfToken);
+    if (!userData) {
+      return returnValue;
+    }
+    this._profileManager.storeData(profileId, AfterDark.name, userData);
+    returnValue.username = userData.username;
+    returnValue.status = LoginStatus.LOGGED_IN;
+    return returnValue;
+  }
+
+  public async post(submission: Submission, postData: SubmissionPostData): Promise<PostResult> {
+    if (submission.submissionType !== SubmissionType.SUBMISSION) {
+      throw new Error('Submissions only.')
+    }
+    if (postData.typeOfSubmission != TypeOfSubmission.ART) {
+      throw new Error('Art submissions only.');
+    }
```

Comment:
Remove.
Do this kind of check in the validation function.

Check how twitter.service.ts checks for file type on line 21 inside of submissionValidate.

Reply:
check

Reply:
Replaced this check with a check in submissionValidate.